ocaml/xapi/import_raw_vdi.ml

```diff
@@ -98,33 +98,51 @@ let import vdi (req: Request.t) (s: Unix.file_descr) _ =
 
   (* Perform the SR reachability check using a fresh context/task because
      	   we don't want to complete the task in the forwarding case *)
-  Server_helpers.exec_with_new_task "VDI.import"
-    (fun __context ->
-       Helpers.call_api_functions ~__context
-         (fun rpc session_id ->
-            let sr_opt = match vdi, sr_of_req ~__context req with
-              | Some vdi, _ -> Some (Db.VDI.get_SR ~__context ~self:vdi)
-              | None, Some sr -> Some sr
-              | None, None -> None
-            in
-            match sr_opt with
-            | Some sr ->
-              debug "Checking whether localhost can see SR: %s" (Ref.string_of sr);
-              if (Importexport.check_sr_availability ~__context sr)
-              then localhost_handler rpc session_id vdi req s
-              else
-                let host = Importexport.find_host_for_sr ~__context sr in
-                let address = Db.Host.get_address ~__context ~self:host in
-                return_302_redirect req s address;
+  try         
+    Server_helpers.exec_with_new_task "VDI.import"
+      (fun __context ->
+         Helpers.call_api_functions ~__context
+           (fun rpc session_id ->
+              let sr_opt = match vdi, sr_of_req ~__context req with
+                | Some vdi, _ -> Some (Db.VDI.get_SR ~__context ~self:vdi)
+                | None, Some sr -> Some sr
+                | None, None -> None
+              in
+              match sr_opt with
+              | Some sr ->
+                debug "Checking whether localhost can see SR: %s" (Ref.string_of sr);
+                if (Importexport.check_sr_availability ~__context sr)
+                then localhost_handler rpc session_id vdi req s
+                else 
+                  let host = Importexport.find_host_for_sr ~__context sr in
+                  let address = Db.Host.get_address ~__context ~self:host in
+                  return_302_redirect req s address;
+                  None
+              | None ->
+                error "Require an SR or VDI to import";
+                let e = Api_errors.(Server_error(vdi_missing,[])) in 
+                let task_id = Xapi_http.task_id_of_req req in
+                begin match task_id with
+                  | Some task_id -> TaskHelper.failed ~__context:(Context.from_forwarded_task task_id) e
+                  | None -> TaskHelper.failed ~__context e
+                end;  
+                Http_svr.headers s (Http.http_400_badrequest ());
                 None
-            | None ->
-              error "Require an SR or VDI to import";
-              TaskHelper.failed ~__context Api_errors.(Server_error(vdi_missing,[]));
-              Http_svr.headers s (Http.http_400_badrequest ());
-              None
-
          )
     )
+  with 
+  | Api_errors.Server_error (a,b) as e ->
```

Comment:
Since `a` and `b` are never used, this could be
`Api_errors.Server_error (_,_)` or even just
`Api_errors.Server_error _`